src/lib/io/Kconfig

```diff
@@ -81,3 +81,28 @@ config USE_MEMMAP
         Users must provide a mapping - for instance, using a JSON
         file - that will be used to define where on memory
         entries will be saved. Examples of memory are NVRAM or EEPROM.
+
+config USE_IPM
+    bool "Inter-processor messaging"
+    depends on FEATURE_HW_IPM
+    default y
+    help
+        Provide API to Inter-processor messaging. Some platforms
+        have more than one processor - this API provides communication
+        among them.
+
+config IPM_CHANNEL_X86_TO_ARC
+    int "IPM Channel x86 -> ARC"
+    depends on USE_IPM
+    default 0
+    help
+        Which IPM channel should be used to communication from x86 core
+        to ARC one.
+
+config IPM_CHANNEL_ARC_TO_X86
+    int "IPM Channel ARC -> x86"
+    depends on USE_IPM
+    default 1
```

Comment:
On 09-05-2016 16:03, Gustavo Sverzut Barbieri wrote:
> In src/lib/io/Kconfig
> <https://github.com/solettaproject/soletta/pull/1996#discussion_r62554193>:
>
>> +        Provide API to Inter-processor messaging. Some platforms
>> +        have more than one processor - this API provides communication
>> +        among them.
>> +
>> +config IPM_CHANNEL_X86_TO_ARC
>> +    int "IPM Channel x86 -> ARC"
>> +    depends on USE_IPM
>> +    default 0
>> +    help
>> +        Which IPM channel should be used to communication from x86 core
>> +        to ARC one.
>> +
>> +config IPM_CHANNEL_ARC_TO_X86
>> +    int "IPM Channel ARC -> x86"
>> +    depends on USE_IPM
>> +    default 1
>
> can't we use the same id for the other way around?

And use one channel? If so, both cores will be interrupted with 
information. Idea is to use one channel for each direction...

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1996/files/64ac7ea3f07d3a26a3d264869f28dc376ba64cc9#r62554193>
>